Tasks/ContainerStructureTestV0/testresultspublisher.ts

```diff
@@ -46,6 +47,12 @@ export class TestResultPublisher {
     public publishToTcm(testResults: TestSummary, testRunTitle: string) {
         let resultsFile = this.createResultsFile(JSON.stringify(testResults));
 
+        const agentVersion = tl.getVariable('Agent.Version');
```

Comment:
does 'minimumAgentVersion ' in task.json not helping?

Reply:
I think that will not allow, the entire task itself. I wan the test to run, but just the publishing will fail. Hence I don't want to fail the task on publishing to TCM itself.

Reply:
Then in that case instead of throwing exception. can we log the warning and pass the task.

Reply:
I am just throwing it from the inner method, in the outer method I just do telemetry and right it as warning.